test/acceptance/test/utils.go

```diff
@@ -337,39 +337,33 @@ func getGitRepoVisibility(org string, repo string, providerName gitproviders.Git
 	return visibility
 }
 
-func waitForResource(resourceType string, resourceName string, namespace string, timeout time.Duration) error {
-	pollInterval := 5
-
-	if timeout < 5*time.Second {
-		timeout = 5 * time.Second
-	}
-
-	timeoutInSeconds := int(timeout.Seconds())
-	for i := pollInterval; i < timeoutInSeconds; i += pollInterval {
-		log.Infof("Waiting for %s in namespace: %s... : %d second(s) passed of %d seconds timeout", resourceType+"/"+resourceName, namespace, i, timeoutInSeconds)
-		err := runCommandPassThroughWithoutOutput([]string{}, "sh", "-c", fmt.Sprintf("kubectl get %s %s -n %s", resourceType, resourceName, namespace))
-
-		if err == nil {
-			log.Infof("%s is available in cluster", resourceType+"/"+resourceName)
-			command := exec.Command("sh", "-c", fmt.Sprintf("kubectl get %s %s -n %s", resourceType, resourceName, namespace))
-			session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
-			Expect(err).ShouldNot(HaveOccurred())
-			Eventually(session).Should(gexec.Exit())
-
-			noResourcesFoundMessage := fmt.Sprintf("No resources found in %s namespace", namespace)
-
-			if strings.Contains(string(session.Wait().Out.Contents()), noResourcesFoundMessage) {
-				log.Infof("Got message => {" + noResourcesFoundMessage + "} Continue looking for resource(s)")
-				continue
-			}
-
-			return nil
+//Assumes resource will eventually contain a status.Conditions where Type=Ready exists
+func waitForResourceToBeReady(resourceType string, resourceName string, namespace string, timeout time.Duration) {
+	EventuallyWithOffset(1, func() error {
+		log.Infof("Waiting for %s/%s in namespace: %q to be ready : ", resourceType, resourceName, namespace)
+		kubectlCommand := fmt.Sprintf("kubectl -n %s wait --for=condition=ready %s %s", namespace, resourceType, resourceName)
+		if resourceName == "" {
+			kubectlCommand = kubectlCommand + " --all"
 		}
+		command := exec.Command("sh", "-c", kubectlCommand)
+		_, err := command.CombinedOutput()
+		if err != nil {
+			return err
+		}
```

Comment:
```suggestion
		if _, err := exec.Command("sh", "-c", kubectlCommand).CombinedOutput(); err != nil {
			return err
		}
```

no need for the extra assignment since it's not used any further